refactor(LicenseActivity): fix material you in license activity, add ApplicationTheme composable that applies correct color scheme (#27)

dbhavsar76 opened the pull request:
# Screenshots

## Different device themes
<img width="300" alt="image" src="https://github.com/user-attachments/assets/86196c92-46e8-489c-9e06-23d4e1dd35c1" />
&nbsp;&nbsp;&nbsp;
<img width="300" alt="image" src="https://github.com/user-attachments/assets/3d49d82e-0151-4d89-9ddf-656e4d50cecc" />

## Dark mode
<img width="300" alt="image" src="https://github.com/user-attachments/assets/9c822aac-2861-492d-a019-f64f33f44e7c" />

## Default theme where material you is not supported (Android 6 Screenshot)
<img width="300" alt="image" src="https://github.com/user-attachments/assets/f381d87c-f2fa-488c-8323-5f0cf795f12b" />
&nbsp;&nbsp;&nbsp;
<img width="300" alt="image" src="https://github.com/user-attachments/assets/9182e711-cc8c-462e-b21a-2bda0a76216b" />

dbhavsar76 committed:
refactor(LicenseActivity): fix material you in license activity, add ApplicationTheme composable that applies correct color scheme

researchxxl commented:
thank you very much for making the pr 🙂

is the settings - behaviour - theme mode the user configures also respected by licenseactivity??

could we please make the diff minimal to allow better code maintenace.. the same content in rows should not jump around in the diff if it can be avoided :))

dbhavsar76 commented:
> is the settings - behaviour - theme mode the user configures also respected by licenseactivity??

Yes, The `LicenseActivity` now extends `ThemedAppCompatActivity`, so the user-selected theme mode (light / dark / follow system) is applied via `AppCompatDelegate.setDefaultNightMode(...)`.
`ApplicationTheme` derives dark/light using `isSystemInDarkTheme()`, which reflects the same configuration.
So behavior is consistent with the rest of the app.

> the same content in rows should not jump around in the diff if it can be avoided :))

The indentation shift is because I removed a redundant `Surface` surrounding the `Scaffold`. And I moved the resource and libraries variables code closer to its usage because its more readable imo (Declaration close to usage and they touch minimal scopes.).

Though I understand keeping the diff minimal for better history maintenance also. I'll push a commit for that.

dbhavsar76 committed:
revert unnecessary code changes

dbhavsar76 commented:
Added new commit to revert unnecessary changes.
Removing unused imports is fine, right?

researchxxl committed:
Merge branch 'main' into refactor-license-activity

researchxxl commented:
@dbhavsar76 many thank you!! :)) perfectly done
